x/slashrefund/keeper/msg_server_test.go

```diff
@@ -1,19 +1,236 @@
 package keeper_test
 
 import (
-	"context"
 	"testing"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
-	banktestutil "github.com/cosmos/cosmos-sdk/x/bank/testutil"
+
+	"github.com/made-in-block/slash-refund/app"
+	"github.com/made-in-block/slash-refund/testutil/testsuite"
 	"github.com/made-in-block/slash-refund/x/slashrefund/keeper"
-	"github.com/made-in-block/slash-refund/x/slashrefund/testslashrefund"
 	"github.com/made-in-block/slash-refund/x/slashrefund/types"
 
+
 	"github.com/stretchr/testify/require"
 )
 
+var (
+	numDelAddrs = 1
+	numValAddrs = 1
+	numDepAddrs = 1
+)
+
+func SetupMsgServerTest() (
+	*app.App, 
+	sdk.Context, 
+	[]sdk.AccAddress, 
+	[]sdk.ValAddress,
+	[]sdk.AccAddress,
+) {
+
+	// Setup delegators
+	delAddrs := testsuite.GenerateNAddresses(numDelAddrs)
+	delAccs := testsuite.ConvertAddressesToAccAddr(delAddrs)
+	balances := testsuite.GenerateBalances(delAccs)
+
+	// Setup validators
+	valAddrs := testsuite.GenerateNAddresses(numValAddrs)
+	valAccs := testsuite.ConvertAddressesToValAddr(valAddrs)
+
+	// Setup depositors
+	depAddrs := testsuite.GenerateNAddresses(numDepAddrs)
+	depAccs := testsuite.ConvertAddressesToAccAddr(depAddrs)
+	depBalances := testsuite.GenerateBalances(depAccs)
+
+	balances = append(balances, depBalances...)
+
+	app, ctx := testsuite.CreateTestApp(delAccs, valAccs, balances, false)
+
+	return app, ctx, delAccs, valAccs, depAccs
+}
+
+// -------------------------------------------------------------------------------------------------
+// Deposit
+// -------------------------------------------------------------------------------------------------
+func TestDepositValid(t *testing.T) {
```

Comment:
Should check which error is returned with require.ErrorIs(), in order to be sure of the actual msgServer behaviour.

Reply:
good point